src/interfaces/chain.cpp

```diff
@@ -268,10 +268,13 @@ class ChainImpl : public Chain
     {
         const BlockFilterIndex* block_filter_index = GetBlockFilterIndex(BlockFilterType::BASIC);
         if (!block_filter_index) return true;
-        const CBlockIndex* index = LookupBlockIndex(hash);
-        if (!index) return true;
         BlockFilter filter;
-        if (!block_filter_index->LookupFilter(index, filter)) return true;
+        {
+            LOCK(cs_main);
+            const CBlockIndex* index = LookupBlockIndex(hash);
+            if (!index) return true;
+            if (!block_filter_index->LookupFilter(index, filter)) return true;
```

Comment:
You don't need the lock for the filter lookup, just the block index lookup.

Reply:
CBlockIndex::nDataPos is protected by cs_main, but we never read that in LookupFilter, so yeah, don't need the lock